src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -269,7 +271,43 @@ public void delete(final String eventTypeName, final Client client)
         }
     }
 
-    private void updateRetentionTime(final String eventTypeName, final Long retentionTime)
+    private void updateEventTypeInDB(final EventType eventType, final Long newRetentionTime,
+                                     final Long oldRetentionTime) throws NakadiException {
+        final NakadiException exception = transactionTemplate.execute(action -> {
+            try {
+                eventTypeRepository.update(eventType);
+                updateTimelinesCleanup(eventType.getName(), newRetentionTime, oldRetentionTime);
```

Comment:
Very good point 👍  I switched the order https://github.com/zalando/nakadi/pull/628/commits/7c4fd7a566ac2dc2a3698de9c096ed06fc239bee
Now the case will be updated with latest state of timelines.